python/deptry/reporters/github.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+from deptry.reporters.base import Reporter
+
+if TYPE_CHECKING:
+    from deptry.violations import Violation
+
+
+@dataclass
+class GithubReporter(Reporter):
+    warning_ids: list[str] = field(default_factory=list)  # list of error codes to print as warnings
+
+    def report(self) -> None:
+        self._log_and_exit()
+
+    def _log_and_exit(self) -> None:
+        self._log_violations(self.violations)
+
+    def _log_violations(self, violations: list[Violation]) -> None:
+        for violation in violations:
+            self._print_github_annotation(violation)
+
+    def _print_github_annotation(self, violation: Violation) -> None:
+        annotation_severity = "warning" if violation.error_code in self.warning_ids else "error"
+        file_name = violation.location.file
+        if violation.location.line is not None and violation.location.column is not None:
+            ret = _build_workflow_command(
+                annotation_severity,
+                str(file_name),
+                violation.location.line,
+                column=violation.location.column,
+                title=violation.error_code,
+                message=violation.get_error_message(),
+            )
+            print(ret)  # noqa: T201
```

Comment:
I wonder if we should use [click.echo](https://click.palletsprojects.com/en/stable/utils/) here instead.

Reply:
done